tests/event.test.ts

```diff
@@ -0,0 +1,248 @@
+    const [event1] = EventFactory.with({ attendanceCode: 'code' });
+    const [event2] = EventFactory.with({ attendanceCode: 'code' });
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createEvents([event1])
+      .createUsers([admin])
+      .write();
+
+    const eventController = ControllerFactory.event(conn);
+
+    await expect(
+      eventController.createEvent({ event: event2 }, admin),
+    ).rejects.toThrow('Attendance code has already been used');
+
+    const eventsResponse = await eventController.getAllEvents({}, admin);
+
+    expect(eventsResponse.events).toHaveLength(1);
+    expect(eventsResponse.events[0]).toStrictEqual(event1.getPublicEvent(true));
+  });
+
+  test('events can be updated by an admin', async () => {
+    const conn = await DatabaseConnection.get();
+    const [event] = EventFactory.with({ attendanceCode: 'code' });
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createEvents([event])
+      .createUsers([admin])
+      .write();
+
+    const eventChanges = {
+      title: 'new title',
+      description: 'new description',
+    };
+
+    const updateEventResponse = await ControllerFactory
+      .event(conn)
+      .updateEvent({ uuid: event.uuid }, { event: eventChanges }, admin);
+
+    expect(updateEventResponse.event).toMatchObject({ ...event.getPublicEvent(), ...eventChanges });
+  });
+
+  test('event attendance code cannot be updated to a duplicate one', async () => {
+    const conn = await DatabaseConnection.get();
+    const [event1] = EventFactory.with({ attendanceCode: 'code' });
+    const [event2] = EventFactory.with({ attendanceCode: 'another-code' });
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createEvents([event1, event2])
+      .createUsers([admin])
+      .write();
+
+    event1.attendanceCode = event2.attendanceCode;
+
+    await expect(
+      ControllerFactory.event(conn).updateEvent({ uuid: event1.uuid }, { event: event1 }, admin),
+    ).rejects.toThrow('Attendance code has already been used');
+  });
+});
+
+describe('event covers', () => {
+  test('properly updates cover photo in database', async () => {
+    const conn = await DatabaseConnection.get();
+    const [event] = EventFactory.create(1);
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const cover = FileFactory.image(Config.file.MAX_EVENT_COVER_FILE_SIZE / 2);
+    const fileLocation = 'fake location';
+
+    await new PortalState()
+      .createUsers([admin])
+      .createEvents([event])
+      .write();
+
+    const eventCoverResponse = await ControllerFactory
+      .event(conn, null, Mocks.storage(fileLocation))
+      .updateEventCover(cover, { uuid: event.uuid }, admin);
+
+    expect(eventCoverResponse.event.cover).toEqual(fileLocation);
```

Comment:
I think we should get the event and check the cover from that instead of checking the `updateEventCoverResponse`.